### ISSUE TITLE
Help: The DDNS updater can't get public ip and still reportting error. 

### ISSUE THREAD

**Issue by code-moritz** [qdm12/ddns-updater#119]
**TLDR**: *Describe your issue in a one liner here*

1. Is this urgent?

    - [x] Yes
    - [ ] No

2. What DNS service provider(s) are you using?

    - [x] Cloudflare
    - [ ] DDNSS.de
    - [ ] DonDominio
    - [ ] DNSOMatic
    - [ ] DNSPod
    - [ ] Dreamhost
    - [ ] DuckDNS
    - [ ] DynDNS
    - [ ] GoDaddy
    - [ ] Google
    - [ ] He.net
    - [ ] Infomaniak
    - [ ] Namecheap
    - [ ] NoIP

3. What's the version of the program?

    **See the line at the top of your logs**

    ```
    Running version latest built on 2020-11-11T22:29:05Z (commit 166b0c7)
    🔧  Need help? https://github.com/qdm12/ddns-updater/issues/new
    💻  Email? quentin.mcgaw@gmail.com
    ☕  Slack? Join from the Slack button on Github
    💸  Help me? https://github.com/sponsors/qdm12
    2020-11-21T16:02:01.237Z	ERROR	IP 1 of 1 is empty for record {"domain":"ws.xxxx.mobi","host":"@","ips":[{"ip":"","time":"2020-11-21T14:12:52.663843986Z"}]}
    ```

4. What are you using to run the container?

    - [ ] Docker run
    - [x] Docker Compose
    - [ ] Kubernetes
    - [ ] Docker stack
    - [ ] Docker swarm
    - [ ] Podman
    - [ ] Other:

5. Extra information

Logs:

```log

```

Configuration file:

```json
{
    "settings": [
        {
                "provider": "cloudflare",
                "zone_identifier":"9d6da6xxxxxxxx336566",
                "domain": "ws.xxxxxxx.mobi",
                "host":"*",
                "ttl":120,
                "email":"jixxxxxx100001@gmail.com",
                "key":"830aexxxxx343a58243fc07f5fd",
                "ip_version":"ipv4 or ipv6"
        }
    ]
}
```

Host OS:
CentOS 8 64bit

**Comment by qdm12** [comment]
That's coming from the json persistent file, there is probably a bad value in there for an IP address. You can try removing updates.json and it should work again.

**Comment by qdm12** [comment]
Please let me know if you have manually modified it or if some bad IP address value was introduced programmatically so I could fix it. Thanks!

**Comment by code-moritz** [comment]
I'am already remove the updates.json file , but the ddns client is still can't work normal. 

The fllowing logs is the latest running info:

============================================================================

Running version latest built on 2020-11-11T22:29:05Z (commit 166b0c7)


🔧  Need help? https://github.com/qdm12/ddns-updater/issues/new
💻  Email? quentin.mcgaw@gmail.com
☕  Slack? Join from the Slack button on Github
💸  Help me? https://github.com/sponsors/qdm12
2020-11-23T13:54:24.649Z        INFO    Found single setting to update record
2020-11-23T13:54:29.652Z        WARN    HTTP GET failed for https://google.com: Get "https://google.com": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
2020-11-23T13:54:29.653Z        WARN    HTTP GET failed for http://google.com: Get "http://google.com": context deadline exceeded (Client.Timeout exceeded while awaiting 
headers)
2020-11-23T13:54:29.653Z        WARN    Domain name resolution is not working for google.com: lookup google.com on 1.1.1.1:53: read udp 172.17.0.2:47768->1.1.1.1:53: i/o 
timeout
2020-11-23T13:54:29.653Z        INFO    Reading history from database: domain ws.xxxxx.mobi host *
2020-11-23T13:54:29.653Z        INFO    Web UI listening at address 0.0.0.0:8001 with root URL ""
2020-11-23T13:54:29.654Z        ERROR   cannot get public ipv4 or ipv6 address: Get "https://diagnostic.opendns.com/myip": dial tcp: lookup diagnostic.opendns.com on 1.1.1.1:53: read udp 172.17.0.2:33723->1.1.1.1:53: read: no route to host
2020-11-23T13:54:29.656Z        INFO    backup: disabled
2020-11-23T13:54:34.657Z        WARN    lookup any.ws.451zone.mobi on 1.1.1.1:53: read udp 172.17.0.2:57192->1.1.1.1:53: read: no route to host

**Comment by qdm12** [comment]
I think the container just doesn't have connectivity to Internet. Are you running some kind of internal firewall?

**Comment by code-moritz** [comment]
I'am not sure.  my host os --- centos can access internet normal. 

Should i open some ports or other configure with docker firewall?

**Comment by qdm12** [comment]
Maybe the DNS `1.1.1.1` it's using to resolve domain names is blocked on your network, I'll check my code this evening so that it uses your DNS and not Cloudflare one.

You can try with

```sh
docker run -it --rm --dns 1.1.1.1 alpine:3.12 wget -O- https://ipinfo.io
```

to see if it's actually blocked.

**Comment by code-moritz** [comment]
[root@localhost containers]# docker run -it --rm --dns 1.1.1.1 alpine:3.12 wget -O- https://ipinfo.io
Unable to find image 'alpine:3.12' locally
3.12: Pulling from library/alpine
188c0c94c7c5: Pull complete
Digest: sha256:c0e9560cda118f9ec63ddefb4a173a2b2a0347082d7dff7dc14272e7841a5b5a
Status: Downloaded newer image for alpine:3.12
wget: bad address 'ipinfo.io'

**Comment by qdm12** [comment]
Yes, it looks blocked. If you try without `--dns 1.1.1.1` I guess it works right? Are you blocking port 53 udp outbound? Or could be your ISP/government. Anyway, I'll make the change, at least so that it's configurable.

**Comment by qdm12** [comment]
Hello there, can you try pulling `:latest` and try again? I'm not sure why it uses the `1.1.1.1` DNS as it's not really enforced as far as I know. If this doesn't work try adding `--dns 8.8.8.8` to your run command so it uses Google's DNS for example instead of `1.1.1.1`.

**Comment by qdm12** [comment]
That seems like a network issue on your side 😕 I cannot find anywhere something specifying the DNS `1.1.1.1`, at least in the current `:latest` image. Maybe try repulling the image as suggested above. I'll close the issue for now but feel free to comment back.
